File: bindal/bindal/custom/js/salary_slip.js
frappe.ui.form.on("Salary Slip", {

    custom_naming_series: function(frm) {
        const branchMapping = {
            'HO-SALSLIP27-.####': 'Head Office',
            'U1-SALSLIP27-.####': 'Unit-1',
            'U2-SALSLIP27-.####': 'Unit-2',
            'U3-SALSLIP27-.####': 'Unit-3',
            'U4-SALSLIP27-.####': 'Unit-4'
        };

        if (branchMapping[frm.doc.custom_naming_series]) {
            frm.set_value('branch', branchMapping[frm.doc.custom_naming_series]);
        }
    }
})

File: bindal/bindal/custom/py/salary_slip.py
import frappe
from frappe.model.naming import make_autoname

def create_name(doc, event):
    if doc.employee:
        branch = frappe.db.get_value("Employee", doc.employee, "branch")

        # Define mapping of branch to naming series
        naming_series_map = {
            "Head Office": "HO-SALSLIP26-.####",
            "Unit-1": "U1-SALSLIP26-.####",
            "Unit-2": "U2-SALSLIP26-.####",
            "Unit-3": "U3-SALSLIP26-.####",
            "Unit-4": "U4-SALSLIP26-.####",
        }

        # Check if branch is in the mapping
        if branch in naming_series_map:
            doc.name = make_autoname(naming_series_map[branch])
        else:
            # fallback naming series if branch doesn't match
            doc.name = make_autoname("SALSLIP26-.####")
